### PR TITLE
Adding `show-kcfg` and `view-kcfg`

### PR DESCRIPTION
**Pull request by dkcumming**
- Refactored the CLI stuff to `cli.py`
- Added `show-kcfg`
- Added `view-kcfg`

### CODE REVIEW

File: kmir/src/kmir/utils.py [hunk 1]
```diff
@@ -16,11 +16,44 @@
 from pyk.proof import APRBMCProof, APRBMCProver, APRProof, APRProver
 from pyk.proof.equality import EqualityProof, EqualityProver
 from pyk.proof.proof import Proof, ProofStatus
-from pyk.utils import BugReport
+from pyk.utils import BugReport, single
+
+T1 = TypeVar('T1')
+T2 = TypeVar('T2')
```
Comment (Baltoli) [inline kmir/src/kmir/utils.py:22]: Can these use the dedicated syntax for type variables? https://docs.python.org/3/library/typing.html#typing.TypeVar

File: kmir/src/kmir/cli.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        )
+        args.add_argument(
+            '--to-module', dest='to_module', default=False, action='store_true', help='Output edges as a K module.'
+        )
+        args.add_argument(
+            '--pending', dest='pending', default=False, action='store_true', help='Also display pending nodes'
+        )
+        args.add_argument(
+            '--failing', dest='failing', default=False, action='store_true', help='Also display failing nodes'
+        )
```
Comment (Baltoli) [inline kmir/src/kmir/cli.py:285]: argparse will infer precisely these names for you, but I presume that you're specifying the `dest` explicitly for consistency with the rest of the options

Comment (Baltoli) [reply kmir/src/kmir/cli.py:285]: (see other places as well for a similar comment)

File: kmir/src/kmir/cli.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        '--save-directory',
+        dest='save_directory',
+        type=dir_path,
+        help='Path to KCFG proofs directory, directory must already exist.',
+    )
+    prove_subparser.add_argument(
+        '--reinit',
+        action='store_true',
+        default=False,
+        help='Reinit a proof.S',
```
Comment (Baltoli) [inline kmir/src/kmir/cli.py:169]: Typo here?
```suggestion
        help='Reinitialize a proof.',
```

File: kmir/src/kmir/cli.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        default='kast',
+        help='Output mode',
+        choices=['pretty', 'program', 'json', 'kore', 'kast', 'none'],
+        required=False,
+    )
+    run_subparser.add_argument(
+        '--ignore-return-code',
+        action='store_true',
+        default=False,
+        help='Ignore return code of krun, alwasys return 0 (use for debugging only)',
```
Comment (Baltoli) [inline kmir/src/kmir/cli.py:90]:
```suggestion
        help='Ignore return code of krun, always return 0 (use for debugging only)',
```

**Review by Baltoli — COMMENTED** [review]
A couple of very minor nitpicks; will leave @geo2a to give a final approval as he's more familiar with the prover code than me!

File: kmir/src/kmir/utils.py (hunk 1 above)
Comment (dkcumming) [reply kmir/src/kmir/utils.py:22]: Our project is using python 3.10, and when I tried to implement the change according to the documentation I ran into this error `Function type parameter syntax requires Python 3.12 or newer Pylance`.

**Review by geo2a — APPROVED** [review]

**Pull request merged**